### PR TITLE
selfcheck: fix summary fields in patron information

### PR DESCRIPTION
**Pull request by lauren-d**
* Adds config to map summary fields by loan state.
* Adapts patron_information method.
* Adds some test.
* Updates invenio-sip2 dependency to v0.6.19.
* Closes rero#3173.

Co-Authored-by: Lauren-D <laurent.dubois@itld-solutions.be>

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* rero/rero-ils-ui#<xx>

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

### CODE REVIEW

**Review by PascalRepond — APPROVED** [review]

**Review by zannkukai — APPROVED** [review]

File: rero_ils/modules/selfcheck/api.py [hunk 1]
```diff
@@ -220,23 +220,17 @@ def patron_information(barcode, **kwargs):
...
+            sip2_summary_fields = current_app.config.get('SIP2_SUMMARY_FIELDS')
+            for loan in get_loans_by_patron_pid(patron.pid, filter_states):
                 item = Item.get_record_by_pid(loan.item_pid)
-                if loan['state'] == LoanState.ITEM_ON_LOAN:
-                    patron_account_information.get('charged_items', []).append(
-                        item.get('barcode'))
-                    if loan.is_loan_overdue():
-                        patron_account_information.get('overdue_items', []) \
+                field = sip2_summary_fields.get(loan['state'])
+                patron_account_information.get(field, []).append(
```
Comment (jma) [inline rero_ils/modules/selfcheck/api.py:227]: You need to use setdefault. If the key does not exits it put the value in a new fresh created array which is not linked to the `patron_account_information`.

File: rero_ils/modules/selfcheck/api.py [hunk 2]
```diff
@@ -220,23 +220,17 @@ def patron_information(barcode, **kwargs):
...
+                    item.get('barcode')
+                )
+                if loan['state'] == LoanState.ITEM_ON_LOAN \
+                        and loan.is_loan_overdue():
+                    patron_account_information.get('overdue_items', []) \
```
Comment (jma) [inline rero_ils/modules/selfcheck/api.py:232]: same as previous

File: tests/api/selfcheck/test_selfcheck.py [hunk 3]
```diff
@@ -190,10 +191,57 @@ def test_patron_information(client, librarian_martigny,
...
     assert res.status_code == 200
+    # create validated request
+    circ_params = {
+        'item_pid': item3_lib_martigny.pid,
+        'patron_pid': selfcheck_patron_martigny.pid,
+        'pickup_location_pid': loc_public_martigny.pid,
+        'transaction_library_pid': lib_martigny.pid,
+        'transaction_user_pid': librarian_martigny.pid
+    }
+    res, data = postdata(
```
Comment (jma) [inline tests/api/selfcheck/test_selfcheck.py:202]: Probably you need to do a cancel request to clean this test.

**Review by jma — CHANGES_REQUESTED** [review]

File: rero_ils/config.py [hunk 4]
```diff
@@ -3395,6 +3395,13 @@ def _(x):
     docmaintype_movie_series='VIDEO',
 )
 
+SIP2_SUMMARY_FIELDS = {
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply rero_ils/config.py:3402]: done

File: tests/api/selfcheck/test_selfcheck.py (hunk 3 above)
Comment (lauren-d) [reply tests/api/selfcheck/test_selfcheck.py:202]: yes of course.

File: rero_ils/modules/selfcheck/api.py [hunk 5]
```diff
@@ -220,23 +220,17 @@ def patron_information(barcode, **kwargs):
...
-            for loan in loans:
+            sip2_summary_fields = current_app.config.get('SIP2_SUMMARY_FIELDS')
+            for loan in get_loans_by_patron_pid(patron.pid, filter_states):
                 item = Item.get_record_by_pid(loan.item_pid)
-                if loan['state'] == LoanState.ITEM_ON_LOAN:
-                    patron_account_information.get('charged_items', []).append(
-                        item.get('barcode'))
-                    if loan.is_loan_overdue():
-                        patron_account_information.get('overdue_items', []) \
+                field = sip2_summary_fields.get(loan['state'])
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply rero_ils/modules/selfcheck/api.py]: done

File: rero_ils/modules/selfcheck/api.py (hunk 2 above)
Comment (lauren-d) [reply rero_ils/modules/selfcheck/api.py]: ok

**Review by jma — APPROVED** [review]

**Pull request merged**